core/domain/collection_domain.py

```diff
@@ -1055,6 +1068,105 @@ def to_dict(self):
             'collection_model_last_updated': self.collection_model_last_updated
         }
 
+    def validate(self):
+        """Validates various properties of the CollectionSummary.
+
+        Raises:
+            ValidationError: One or more attributes of the CollectionSummary
+                are invalid.
+        """
+        if not isinstance(self.title, basestring):
+            raise utils.ValidationError(
+                'Expected title to be a string, received %s' % self.title)
+        utils.require_valid_name(
+            self.title, 'the collection title', allow_empty=True)
```

Comment:
Er, why is allow_empty true for most of these? I thought we only had collection summaries for published collections, which should have non-empty values for most of these.

Reply:
We have collection summaries for unpublished collections as well. When we create a collection, we are creating a summary as well. ([Ref](https://github.com/oppia/oppia/blob/develop/core/domain/collection_services.py#L772))